### PR TITLE
#3131833 - Check whether the order item have options field

### PR DESCRIPTION
**Pull request by safalliajoseph**
Update the options field widget to display all selected options

### CODE REVIEW

**Comment by safalliajoseph** [comment]
@shabananavas This MR just adds a condition to check whether the order item have options field before merging the order items and it also modifies the code to display all selected variation options in the field widget. Please review. Thanks!

**Comment by safalliajoseph** [comment]
https://www.drupal.org/project/3111714/issues/3131833 - d.o issue

**Comment by safalliajoseph** [comment]
https://www.drupal.org/project/3111714/issues/3131833#comment-13577011 - d.o patch

File: src/Resolver/OptionResolver.php [hunk 1]
```diff
@@ -19,32 +21,40 @@ class OptionResolver implements OptionResolverInterface {
...
    * Constructs a new OptionResolver object.
    *
    * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
    *   The entity manager.
+   * @param \Drupal\commerce_option\ProductOptionFieldManager $field_manager
+   *   The option field manager.
    */
-  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+  public function __construct(
+    EntityTypeManagerInterface $entity_type_manager,
```
Comment (shabananavas) [inline src/Resolver/OptionResolver.php]: Do we need entity type manager now in this class?

File: commerce_option.services.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ services:
     arguments: ['@entity_field.manager', '@entity_type.bundle.info', '@entity_type.manager', '@cache.data']
   commerce_option.option_resolver:
     class: Drupal\commerce_option\Resolver\OptionResolver
-    arguments: ['@entity_type.manager']
+    arguments: ['@entity_type.manager', '@commerce_option.option_field_manager']
```
Comment (shabananavas) [inline commerce_option.services.yml]: This is excellent that you're using the field manager instead to grab the option values. Great understanding of the code!

**Comment by shabananavas** [comment]
Everything looks great! Just had one minor comment.

File: src/Resolver/OptionResolver.php (hunk 1 above)
Comment (safalliajoseph) [reply src/Resolver/OptionResolver.php]: Yeah its not needed anymore. I have removed it now

File: commerce_option.services.yml (hunk 2 above)
Comment (safalliajoseph) [reply commerce_option.services.yml]: Yep, We did not had to query db for getting the options info.

**Comment by safalliajoseph** [comment]
@shabananavas I have addressed your feedbacks, the new patch is https://www.drupal.org/project/3111714/issues/3131833#comment-13577066

**Comment by shabananavas** [comment]
Looks good. I'll mark it as RTBC. Please have @thenchev merge it in.

**Comment by safalliajoseph** [comment]
Thank you

**Pull request merged**